components/permissions/permission_origin_lifetime_monitor.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PERMISSIONS_PERMISSION_ORIGIN_LIFETIME_MONITOR_H_
+#define BRAVE_COMPONENTS_PERMISSIONS_PERMISSION_ORIGIN_LIFETIME_MONITOR_H_
+
+#include <string>
+
+#include "base/callback.h"
+#include "url/gurl.h"
+
+namespace permissions {
+
+// Helper to support an origin-based permission lifetime logic.
+class PermissionOriginLifetimeMonitor {
+ public:
+  virtual ~PermissionOriginLifetimeMonitor() = default;
+
+  // Set a callback to call when a permission origin is destroyed.
+  // Callback will receive a string returned by
+  // |SubscribeToPermissionOriginDestruction|.
+  virtual void SetOnPermissionOriginDestroyedCallback(
+      base::RepeatingCallback<void(const std::string&)> callback) = 0;
+
+  // Subscribe to a permission origin destruction. Returns a string key which
+  // will be used in a callback. An empty key can be returned if a permission
```

Comment:
not really clear what is the key

Reply:
made it clearer in the interface implementation which works with ephemeral storage domain directly.

https://github.com/brave/brave-core/blob/ede10c1ef5661ce11977b1f83d81e59e39535fbd/browser/permissions/brave_permission_origin_lifetime_monitor.h#L35-L38